src/org/exist/xquery/PathExpr.java

```diff
@@ -297,7 +297,7 @@ public Sequence eval(Sequence contextSequence, final Item contextItem) throws XP
 
             final boolean allowMixedNodesInReturn;
             if(prev != null) {
-                allowMixedNodesInReturn = prev.allowMixedNodesInReturn() | expr.allowMixedNodesInReturn();
+                allowMixedNodesInReturn = prev.allowMixedNodesInReturn() /*| expr.allowMixedNodesInReturn()*/;
```

Comment:
@wolfgangmm it is unclear why this line needed to be changed for the test to succeed.

Reply:
@adamretter running the test on my install, above comment was not needed. The test passed without it. 

I did not contribute to the code calling allowMixedNodesInReturn, but I can see some sense in it. It's not vital though. I would thus say, if your test passes without commenting out the test above, we should leave PathExpr untouched. If it gets in your way though, we need to reconsider.

Reply:
@wolfgangmm Sure the contributed test is fine. But one of the existing tests in the test suite fails without commenting that out (due to my override in `Function.java`).

Reply:
Ok, so the error expected by `XQueryTest#pathOperatorContainingNodesAndNonNodes()` is in line with what saxon reports. I wonder why you had to overwrite `Function#allowMixedNodesInReturn()`? It does not seem related to the original issue.

Reply:
Discard the question, I see it. This is quite a puzzle ...